packages/compass-data-modeling/src/utils/schema-traversal.tsx

```diff
@@ -411,3 +590,34 @@ export function getSchemaWithNewTypes(
   }
   return { anyOf: newVariants };
 }
+
+/**
+ * Gets the direct children of a MongoDB JSON schema.
+ * @param field - field to get direct children for
+ * @returns direct children of the field (if any)
+ */
+export const getDirectChildren = (
+  schema: MongoDBJSONSchema
+): [string, MongoDBJSONSchema][] => {
+  // children are either direct (properties), from anyOf, items or items.anyOf
+  const children: [string, MongoDBJSONSchema][] = [];
+  if (schema.properties) {
+    children.push(...Object.entries(schema.properties));
+  }
+  if (schema.items) {
+    if (!Array.isArray(schema.items)) {
+      children.push(...getDirectChildren(schema.items));
+    } else {
+      for (const item of schema.items) {
+        children.push(...getDirectChildren(item));
+      }
+    }
+  }
+  if (schema.anyOf) {
+    for (const variant of schema.anyOf) {
+      children.push(...getDirectChildren(variant));
+    }
+  }
+
+  return children;
+};
```

Comment:
Optional suggestion, but this is the type of code that yells "iterator" at me 🙂 

```suggestion
export function* getDirectChildren(
  schema: MongoDBJSONSchema
): Iterable<[string, MongoDBJSONSchema]> {
  // children are either direct (properties), from anyOf, items or items.anyOf
  if (schema.properties) {
    yield* Object.entries(schema.properties);
  }
  if (schema.items) {
    if (!Array.isArray(schema.items)) {
      yield* getDirectChildren(schema.items);
    } else {
      for (const item of schema.items) {
        yield* getDirectChildren(item);
      }
    }
  }
  if (schema.anyOf) {
    for (const variant of schema.anyOf) {
      yield* getDirectChildren(variant);
    }
  }
}
```